frontend/src/preferences/preferenceApi.ts

```diff
@@ -46,4 +46,20 @@ export function setLocalApiKey(key: string): void {
 
 export function setLocalRecentDocs(doc: Doc): void {
   localStorage.setItem('DocsGPTRecentDocs', JSON.stringify(doc));
+  let namePath = doc.name;
+  if (doc.language === namePath) {
+    namePath = '.project';
+  }
+
+  const docPath =
+    doc.language + '/' + namePath + '/' + doc.version + '/' + doc.model;
+  fetch('https://docsgpt.arc53.com/api/docs_check', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      docs: docPath,
+    }),
+  }).then((response) => response.json());
```

Comment:
is this sequence of operation correct?

user selects a document > save the doc to LS > make API request to check if Doc is available

I am not sure about the details here but I feel because theres no error handling in http req there can be a case where LS and backend might be in an inconsistent state.

example :
Do we still want to keep it saved in LS when http req has failed?

If the current implementation does the job we can keep it the way it is now and think about it few days later 😉

Reply:
Agreed need to handle errors here and loading states